syncers/auth_history_syncer/src/main/java/com/yahoo/athenz/syncer/auth/history/AuthHistorySyncer.java

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ *  * Copyright The Athenz Authors
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.yahoo.athenz.syncer.auth.history;
+
+import com.yahoo.athenz.auth.PrivateKeyStore;
+import com.yahoo.athenz.auth.PrivateKeyStoreFactory;
+import com.yahoo.athenz.syncer_common.Config;
+import com.yahoo.athenz.syncer_common.SyncTimeRange;
+import com.yahoo.athenz.syncer_common.Syncer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class AuthHistorySyncer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthHistorySyncer.class);
+    private static final Long MAX_SYNC_RANGE = TimeUnit.MINUTES.toMillis(30);
+    private static final String PROP_REGION = "auth_history_syncer.aws.region";
+    private static final String PROP_REGION_DEFAULT = "us-west-2";
+    private static final String PROP_AUTH_HISTORY_FETCH_FACTORY_CLASS = "auth_history_syncer.fetch_factory_class";
+    private static final String PROP_AUTH_HISTORY_SEND_FACTORY_CLASS = "auth_history_syncer.send_factory_class";
+    private static final String PROP_AUTH_HISTORY_SEND_FACTORY_CLASS_DEFAULT = "com.yahoo.athenz.syncer.auth.history.impl.DynamoDBAuthHistorySenderFactory";
+    private static final String PROP_AUTH_HISTORY_PKEYSTORE_FACTORY_CLASS = "auth_history_syncer.private_key_store_factory_class";
+
+
+    public static void main(String[] args) {
+        if (args.length != 1) {
+            usage();
+            System.exit(1);
+        }
+        Config.loadProperties(args[0]);
+        Syncer syncer = new Syncer();
+        AuthHistorySyncer authHistorySyncer = new AuthHistorySyncer();
+        syncer.sync(authHistorySyncer, (authSyncer, timeRange) -> authSyncer.sync(timeRange));
+    }
+
+    public boolean sync(SyncTimeRange syncTimeRange) {
+        try {
+            // Validate syncTimeRange
+            if (syncTimeRange.getEndTime() - syncTimeRange.getStartTime() > MAX_SYNC_RANGE) {
+                // If no successful run recorded, or time since last run is too big, sync for MAX_SYNC_RANGE milliseconds
+                syncTimeRange.setStartTime(syncTimeRange.getEndTime() - MAX_SYNC_RANGE);
+            }
+
+            LOGGER.info("Start syncing. Range - " + syncTimeRange);
```

Comment:
we should use {} for variables and not +

Reply:
fixed